homeassistant/components/html5/notify.py

```diff
@@ -38,33 +38,21 @@
 from homeassistant.util import ensure_unique_string
 from homeassistant.util.json import JsonObjectType, load_json_object
 
-from .const import DOMAIN, SERVICE_DISMISS
+from .const import (
+    ATTR_VAPID_EMAIL,
+    ATTR_VAPID_PRV_KEY,
+    ATTR_VAPID_PUB_KEY,
+    DOMAIN,
+    SERVICE_DISMISS,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 REGISTRATIONS_FILE = "html5_push_registrations.conf"
 
-ATTR_VAPID_PUB_KEY = "vapid_pub_key"
-ATTR_VAPID_PRV_KEY = "vapid_prv_key"
-ATTR_VAPID_EMAIL = "vapid_email"
-
-
-def gcm_api_deprecated(value):
-    """Warn user that GCM API config is deprecated."""
-    if value:
-        _LOGGER.warning(
-            "Configuring html5_push_notifications via the GCM api"
-            " has been deprecated and stopped working since May 29,"
-            " 2019. Use the VAPID configuration instead. For instructions,"
-            " see https://www.home-assistant.io/integrations/html5/"
-        )
-    return value
-
 
 PLATFORM_SCHEMA = NOTIFY_PLATFORM_SCHEMA.extend(
     {
-        vol.Optional("gcm_sender_id"): vol.All(cv.string, gcm_api_deprecated),
-        vol.Optional("gcm_api_key"): cv.string,
```

Comment:
So these are deprecated, but should we keep the schema intact? Otherwise the schema would fail. But we can warn people with a repair issue when setting up that it doesn't work. I'd rather do that than break systems

Reply:
These were deprecated over 5 years ago or something to that magnitude. Gcm broke long ago and no longer worked for years. If it was there, it definitely doesn't work today.

Reply:
Oh I get that, but it's more that, if people still had this setup, it would raise an exception after this PR. And I would much rather just try importing it, and let the user know that this is removed and that they should use VAPID instead.

Reply:
If that's the case, I could disable config validation to make it work no matter what, but that just doesn't seem to be how home assistant was made to work. Either the config is valid or it isn't. If it isn't, an exception is created.

Reply:
But can we make this a thing before the import? Check if the user has this set and no vapid key, we raise an issue telling them that we can't import the configuration because there was no vapid key found but old gcm? This way people get to remove the yaml and know why it doesn't import

Reply:
Going to revise my thoughts here. People don't really need to know the reason of the import fail other than it failed. The only reason we import in the first place is to keep existing html5 notifications working. Gcm didn't work previously so there is nothing to import. People should expect an error, remove the yaml, and resetup in config flow. Good news is this removes the previous docs which has users go in a roundabout manner to generate keys and does this automatically, reducing friction.

In short, I think it should be left as is.

Reply:
According to ADR-21 we do have to tell the user what failed, and there are only 2 ways it can fail, so I don't think it's that much extra effort. If you want, I can also take a stab at it

Reply:
I read [adr-21](https://github.com/home-assistant/architecture/blob/master/adr/0021-YAML-integration-configuration-deprecation-policy.md) and don't see it saying we need to tell users to that kind of clarity. It just says we need to tell the user what happened (import failed) and what they should do (resetup in config flow) which this current does. From a user standpoint, that's all I really care about anyways. 

If you would like to try your hand at this, feel free.

Reply:
The previous deprecation policy was 2 releases. Adr-21 states the deprecation policy should be 6 releases. This is well outside of both time-frames combined. AND this assumes the user has also ignored the warnings showing up in their logs for the past FIVE years and never updated their config.

Reply:
I can get your frustration, and I'm sad to tell you that these people exist 🥲

I'll take a stab at it this weekend :)

Reply:
Sure they might exist, but why even have ADR if we aren't going to follow it and support way outside the support timeline? It doesn't make sense to keep supporting legacy forever.

Reply:
The ADR only says the lower bounds, not the upper bounds. So I'm aiming to remove that at the same time as the YAML import, does that sound reasonable?

Reply:
I'm still not totally onboard. It really just seems like extra complexity for practically 0 gain. Even if said people exist, their setup was already broken and they have a path forward to fix the setup with the current code.